Math error in GeoHashQuery.java cause code to give unexpected exception

Hello, I'm new to submitting issues to GitHub, but this has to be fixed I think.

I was getting mad on trying to figure out why my app was giving me an Exception when I was setting the GeoQuery radius to a big number (>500'000), so I started debugging the code and found what it seems to me to be a math error.

The Exception that I get is :
` java.lang.IllegalArgumentException: Precision of GeoHash must be larger than zero! 
at com.firebase.geofire.core.GeoHash.<init>(GeoHash.java:29)`

The code I try to execute is:
`someGeoQuery.setRadius(500000);
`
The lines that I think are wrong are:  ( in file **com.firebase.geofire.core.GeoHashQuery.java**)
```
        int queryBits = Math.max(1, Utils.bitsForBoundingBox(location, radius));
        int geoHashPrecision = (int)(Math.ceil(queryBits/Base32Utils.BITS_PER_BASE32_CHAR));
```

In fact, queryBits and Base32Utils.BITS_PER_BASE32_CHAR , are both integer, so a division of that kind can only bring to an integer result, so why the need of the ceil?
I think this is an error, and it cause a consequentially wanted Exception.

The code has to be corrected to :
```
        **float** queryBits = Math.max(1, Utils.bitsForBoundingBox(location, radius));
        int geoHashPrecision = (int)(Math.ceil(queryBits/Base32Utils.BITS_PER_BASE32_CHAR));
```

The version that I'm using is:
`   'com.firebase:geofire-android:2.1.0'
`

Am I right?

It's giving me error:

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] geofire ............................................ SUCCESS [  1.728 s]
[INFO] geofire-android .................................... FAILURE [  4.755 s]
[INFO] geofire-java ....................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:43 min
[INFO] Finished at: 2016-09-22T17:27:36+02:00
[INFO] Final Memory: 18M/121M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project geofire-android: Could not resolve dependencies for project com.firebase:geofire-android:aar:2.1.1-SNAPSHOT: Failed to collect dependencies at com.google.firebase:firebase-database:aar:[9.0.2,): No versions available for com.google.firebase:firebase-database:aar:[9.0.2,) within specified range -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :geofire-android

Do I have to configure maven repos?

Can you check that your Play Services SDK is up to date:
http://stackoverflow.com/questions/37310188/failed-to-resolve-com-google-firebasefirebase-core9-0-0/37310513#37310513

also make sure that the ANDROID_HOME variable in your environment is set to
the location of your android SDK.  For me it looks like this:

$ echo $ANDROID_HOME
/usr/local/google/home/samstern/android-sdk-linux

On Thu, Sep 22, 2016 at 8:30 AM giacomodeglinnocenti <
notifications@github.com> wrote:

> It's giving me error:
>
> [INFO]
> ------------------------------------------------------------------------
> [INFO] Reactor Summary:
> [INFO]
> [INFO] geofire ............................................ SUCCESS [
> 1.728 s]
> [INFO] geofire-android .................................... FAILURE [
> 4.755 s]
> [INFO] geofire-java ....................................... SKIPPED
> [INFO]
> ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO]
> ------------------------------------------------------------------------
> [INFO] Total time: 01:43 min
> [INFO] Finished at: 2016-09-22T17:27:36+02:00
> [INFO] Final Memory: 18M/121M
> [INFO]
> ------------------------------------------------------------------------
> [ERROR] Failed to execute goal on project geofire-android: Could not
> resolve dependencies for project
> com.firebase:geofire-android:aar:2.1.1-SNAPSHOT: Failed to collect
> dependencies at com.google.firebase:firebase-database:aar:[9.0.2,): No
> versions available for com.google.firebase:firebase-database:aar:[9.0.2,)
> within specified range -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the
> -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions,
> please read the following articles:
> [ERROR] [Help 1]
> http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
> [ERROR]
> [ERROR] After correcting the problems, you can resume the build with the
> command
> [ERROR] mvn -rf :geofire-android
>
> Do I have to configure maven repos?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/firebase/geofire-java/issues/46#issuecomment-248938242>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIEw6s5a_sHASyIL9mxiinZFxX6HyVd7ks5qsp76gaJpZM4KAlNV>
> .
>

Hello,
 I've successfully launched the test, before the modification, but it is failing due to some timeout.
Do I have to set a timeout variable at any level?

`[...]
  Failed tests:   geoFireSetsLocations(com.firebase.geofire.GeoFireTest): Timeout occured!
  getLocationReturnsCorrectLocation(com.firebase.geofire.GeoFireTest): Timeout occured!
  getLocationOnWrongDataReturnsError(com.firebase.geofire.GeoFireTest): Timeout occured!
  keyMoved(com.firebase.geofire.GeoQueryTest): Timeout occured!
  keyEntered(com.firebase.geofire.GeoQueryTest): Timeout occured!
  keyExited(com.firebase.geofire.GeoQueryTest): Timeout occured!
  subQueryTriggersKeyMoved(com.firebase.geofire.GeoQueryTest): Timeout occured!
  removeSingleObserver(com.firebase.geofire.GeoQueryTest): Timeout occured!
  removeAllObservers(com.firebase.geofire.GeoQueryTest): Timeout occured!
  readyListener(com.firebase.geofire.GeoQueryTest): Timeout occured!
  readyListenerAfterReady(com.firebase.geofire.GeoQueryTest): Timeout occured!
  readyAfterUpdateCriteria(com.firebase.geofire.GeoQueryTest): Timeout occured!

Tests run: 28, Failures: 12, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:12 min
[INFO] Finished at: 2016-09-26T16:47:23+02:00
[INFO] Final Memory: 15M/195M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project geofire-java: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/jack/Documents/Android/geofire/java/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureExceptio`

--- a/src/main/java/com/firebase/geofire/core/GeoHashQuery.java
+++ b/src/main/java/com/firebase/geofire/core/GeoHashQuery.java
@@ -69,7 +69,7 @@ public class GeoHashQuery {
 
     public static Set<GeoHashQuery> queriesAtLocation(GeoLocation location, double radius) {
         int queryBits = Math.max(1, Utils.bitsForBoundingBox(location, radius));
-        int geoHashPrecision = (int)(Math.ceil(queryBits/Base32Utils.BITS_PER_BASE32_CHAR));
+        int geoHashPrecision = (int)(Math.ceil(((float)queryBits)/Base32Utils.BITS_PER_BASE32_CHAR));
 
         double latitude = location.latitude;
         double longitude = location.longitude;

